src/Sylius/Component/Core/Currency/ChannelAwareCurrencyProvider.php

```diff
@@ -48,4 +49,18 @@ public function getAvailableCurrencies()
             return $currency->isEnabled();
         });
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBaseCurrency()
+    {
+        $currentChannel =  $this->channelContext->getChannel();
+
+        foreach ($currentChannel->getCurrencies() as $currency) {
```

Comment:
I think this is wrong.

You wouldn't get the base currency from the context, you'll get it from the repository. The current customer may have a context with a different currency.